internal/approver/controller/controller_test.go

```diff
@@ -49,21 +48,21 @@ func Test_Reconcile(t *testing.T) {
 	)
 
 	tests := map[string]struct {
-		existingObjects []runtime.Object
-		evaluator       evaluator.Interface
-		expResult       ctrl.Result
-		expError        bool
-		expObjects      []runtime.Object
+		existingCRObjects []client.Object
+		evaluator         evaluator.Interface
+		expResult         ctrl.Result
+		expError          bool
+		expObjects        []client.Object
 	}{
 		"if CertificateRequest doesn't exist, ignore": {
-			existingObjects: []runtime.Object{},
-			expResult:       ctrl.Result{},
-			expError:        false,
-			evaluator:       nil,
-			expObjects:      []runtime.Object{},
+			existingCRObjects: []client.Object{},
```

Comment:
The function to track objects with subresources expects `client.Object`. This does not matter for us as a `CertificateRequest` is a `client.Object` and `client.Object` embeds `runtime.Object`